src/base/api.lua

```diff
@@ -1148,8 +1148,12 @@
 ---
 
 	function configuration(terms)
+		local caller =  filelineinfo(2)
 		-- Sep 16 2021
-		premake.warnOnce("configuration", "`configuration` has been deprecated; use `filter` instead (https://premake.github.io/docs/Filters/)")
+		premake.warnOnce("configuration", "`configuration` has been deprecated; use `filter` instead (https://premake.github.io/docs/Filters/) @%s\n", caller)
+		if api._deprecations == "error" then
+			error("deprecation errors enabled", 3)
```

Comment:
I think 5 years might be long enough for the deprecation, you can probably just remove the function now...